src/printer/smt2/smt2_printer.cpp

```diff
@@ -196,6 +197,19 @@ void Smt2Printer::toStream(std::ostream& out,
         n.constToStream(out);
       }
       break;
+    case kind::ABSTRACT_TYPE:
+    {
+      const AbstractType& at = n.getConst<AbstractType>();
+      Kind atk = at.getKind();
+      out << "?";
+      // note that the fully abstract type (where atk is ABSTRACT_TYPE) is
+      // printed simply as "?", not e.g. "?Abstract"
```

Comment:
```suggestion
      // printed simply as "?", not, e.g., "?Abstract"
```

Reply:
What about abstract types with indices?

Reply:
The "index" of an abstract type is a kind of sort, so we print the name. E.g. `ABSTRACT_TYPE{BITVECTOR_TYPE}` is printed as `?BitVec`.  I will comment on this.